crates/walrus-service/src/client.rs

```diff
@@ -763,28 +792,91 @@ impl<T> Client<T> {
+            config.min_backoff,
+            config.max_backoff,
+            config.max_retries,
+            u64::from_le_bytes(
+                blob_id.0[..8]
+                    .try_into()
+                    .expect("can always convert 8 bytes into u64"),
+            ),
+        );
+        while let Some(delay) = backoff.next_delay() {
+            let maybe_status = self
+                .get_verified_blob_status(blob_id, read_client, delay)
+                .await;
+
+            // Return only if we know that the blob does not exist, or if we have a valid status.
+            match maybe_status {
+                Ok(_) => {
+                    return maybe_status;
+                }
+                Err(client_error)
+                    if matches!(client_error.kind(), &ClientErrorKind::BlobIdDoesNotExist) =>
+                {
+                    return Err(client_error)
+                }
+                Err(_) => (),
+            }
+
+            // TODO(giac): next PR fixes that we wait even if we are out of retries.
+            tracing::debug!(?delay, "fetching blob status failed; retrying");
+            tokio::time::sleep(delay).await;
+        }
+        todo!()
+    }
+
     /// Gets the blob status from multiple nodes and returns the latest status that can be verified.
     ///
-    /// The nodes are selected such that at least one correct node is contacted.
+    /// The nodes are selected such that at least one correct node is contacted. This function reads
+    /// from the latest committee, because, during epoch change, it is the committee that will have
+    /// the most up-to-date information on the old and newly certified blobs.
+    #[tracing::instrument(skip_all, fields(%blob_id), err(level = Level::WARN))]
     pub async fn get_verified_blob_status<U: ReadClient>(
         &self,
         blob_id: &BlobId,
         read_client: &U,
         timeout: Duration,
     ) -> ClientResult<BlobStatus> {
-        let comms = self.node_read_communications()?;
+        tracing::debug!(?timeout, "trying to get blob status");
+        let comms = self
+            .communication_factory
+            .node_read_communications(self.committees.write_committee().epoch)?;
+
         let futures = comms
             .iter()
             .map(|n| n.get_blob_status(blob_id).instrument(n.span.clone()));
         let mut requests = WeightedFutures::new(futures);
         requests
             .execute_until(
-                &|weight| self.committees.current_committee().is_quorum(weight),
+                &|weight| self.committees.is_quorum(weight),
                 timeout,
                 self.communication_limits.max_concurrent_status_reads,
             )
             .await;
 
+        // If 2f+1 nodes return a 404 status, we know the blob does not exist.
+        let n_not_found = requests
+            .inner_err()
+            .iter()
+            .filter(|err| err.is_status_not_found())
+            .count();
```

Comment:
@giac-mysten I know this is a very old PR, but I just stumbled over this code here: Shouldn't we sum up the *weight* and check if that is a quorum instead of counting the *nodes*?